### PR TITLE
drivers: modem: hl7800: Make RX net_bufs configurable

### PR DESCRIPTION
**Pull request by rerickson1**
Make receive network buffer count and size
configurable via kconfig.

### CODE REVIEW

File: drivers/modem/hl7800.c [hunk 1]
```diff
@@ -3117,7 +3115,7 @@ static inline struct net_buf *read_rx_allocator(k_timeout_t timeout,
 
 static size_t hl7800_read_rx(struct net_buf **buf)
 {
-	uint8_t uart_buffer[MDM_RECV_BUF_SIZE];
+	uint8_t uart_buffer[CONFIG_MODEM_HL7800_RECV_BUF_SIZE];
```
Comment (jukkar) [inline drivers/modem/hl7800.c:3118]: BTW, you could avoid this and possible stack issues by reading data directly to net_buf. Having this intermediate stack buffer is not really needed.

**Review by jukkar — APPROVED** [review]
The change looks ok, but I would suggest you to refactor the reading side a bit (in a seprate PR)

File: drivers/modem/hl7800.c (hunk 1 above)
Comment (rerickson1) [reply drivers/modem/hl7800.c:3118]: Looked at this and I am not sure how to accomplish it. `mdm_receiver_recv` needs to read to a buffer and then that buffer is used for `net_buf_append_bytes`. I don't think it's possible to do this another way without significant changes.

**Comment by rerickson1** [comment]
@mike-scott can I get a review when you get a chance?

File: drivers/modem/hl7800.c (hunk 1 above)
Comment (jukkar) [reply drivers/modem/hl7800.c:3118]: There was a patch few days ago that did similar things (avoiding extra temp buffer read), see commit 318cbe649aed5c2a28dea3ceaff7ccd2af871ab4 for details how it did it.

**Review by rlubos — APPROVED** [review]

File: drivers/modem/hl7800.c (hunk 1 above)
Comment (rerickson1) [reply drivers/modem/hl7800.c:3118]: perfect, thank you. I have made note of this and will look at improving this in a new PR.

**Review by mike-scott — APPROVED** [review]

**Pull request merged**